pkg/reconciler/instances/istio/bootstrap.go

```diff
@@ -101,17 +101,31 @@ func parsePaths(input string, isValid func(string) error) ([]string, error) {
 }
 
 // TODO: test it
-func validatePath(path string) error {
+func validatePath(path string, logger *zap.SugaredLogger) error {
 	stat, err := os.Stat(path)
 	if err != nil {
 		return errors.Wrap(err, "Error getting file data")
 	}
 	mode := stat.Mode()
+	logger.Debugf("%s mode: %s", path, mode)
 	if (!mode.IsRegular()) || mode.IsDir() {
 		return errors.New(fmt.Sprintf("\"%s\" is not a regular file", path))
 	}
 	if uint32(mode&0111) == 0 {
-		return errors.New(fmt.Sprintf("\"%s\" is not executable", path))
+		logger.Debugf("%s is not executable, will chmod +x", path)
+		var fileMode os.FileMode = 0777
```

Comment:
After you really extended the logic of the function could you provide some tests? We lack them here

Reply:
Will do so